### ISSUE TITLE
MDCList: List items with checkbox/radio buttons should use aria roles on li since the controls are not focusable

### ISSUE THREAD

**Issue by williamernest** [material-components/material-components-web#3659]
The screen reader does not read the state of the checkbox since the checkbox is not focusable. The list item should contain `role="checkbox|radio"` and should toggle the `aria-checked` property to match the checkbox state.
